ale_linters/mercury/mmc.vim

```diff
@@ -0,0 +1,44 @@
+" Author: stewy33 <slocumstewy@gmail.com>
+" Description: Lints mercury files using mmc
+
+if !exists('g:ale_mercury_mmc_executable')
+    let g:ale_mercury_mmc_executable = 'mmc'
+endif
+
+if !exists('g:ale_mercury_mmc_options')
+    let g:ale_mercury_mmc_options = ''
+endif
+
+function! ale_linters#mercury#mmc#GetCommand(buffer) abort
```

Comment:
Write a `GetExecutable` function which returns the configurable executable, and use it as `executable_callback` below. Call `GetExecutable` here. ALE checks if an executable is executable before running commands, so you need to tell it what the configured executable name is, separately from the command that is run. You should use `ale#Escape()` on the executable name, as it could contain spaces or other invalid characters if it is configurable.

Add a test for both of these functions in the `test/command_callback` directory, which contains a few examples.